airbyte-oauth/src/main/java/io/airbyte/oauth/flows/MicrosoftTeamsOAuthFlow.java

```diff
@@ -0,0 +1,175 @@
+        "User.Read.All",
+        "User.ReadWrite.All");
+  }
+
+  @Override
+  @Deprecated
+  public Map<String, Object> completeSourceOAuth(final UUID workspaceId,
+                                                 final UUID sourceDefinitionId,
+                                                 final Map<String, Object> queryParams,
+                                                 final String redirectUrl)
+      throws IOException {
+    throw new IOException("not supported");
+  }
+
+  @Override
+  @Deprecated
+  public Map<String, Object> completeDestinationOAuth(final UUID workspaceId,
+                                                      final UUID destinationDefinitionId,
+                                                      final Map<String, Object> queryParams,
+                                                      final String redirectUrl)
+      throws IOException {
+    throw new IOException("not supported");
+  }
+
+  @Override
+  public Map<String, Object> completeSourceOAuth(final UUID workspaceId,
+                                                 final UUID sourceDefinitionId,
+                                                 final Map<String, Object> queryParams,
+                                                 final String redirectUrl,
+                                                 final JsonNode inputOAuthConfiguration,
+                                                 final OAuthConfigSpecification oAuthConfigSpecification)
+      throws IOException, ConfigNotFoundException, JsonValidationException {
+    tenantId = null;
+    try {
+      tenantId = getConfigValueUnsafe(inputOAuthConfiguration, fieldName);
+    } catch (final IllegalArgumentException e) {}
+    return super.completeSourceOAuth(workspaceId, sourceDefinitionId, queryParams, redirectUrl, inputOAuthConfiguration, oAuthConfigSpecification);
+  }
+
+  @Override
+  public Map<String, Object> completeDestinationOAuth(final UUID workspaceId,
+                                                      final UUID destinationDefinitionId,
+                                                      final Map<String, Object> queryParams,
+                                                      final String redirectUrl,
+                                                      final JsonNode inputOAuthConfiguration,
+                                                      final OAuthConfigSpecification oAuthConfigSpecification)
+      throws IOException, ConfigNotFoundException, JsonValidationException {
+    tenantId = null;
+    try {
+      tenantId = getConfigValueUnsafe(inputOAuthConfiguration, fieldName);
+    } catch (final IllegalArgumentException e) {}
+    return super.completeDestinationOAuth(workspaceId, destinationDefinitionId, queryParams, redirectUrl, inputOAuthConfiguration,
+        oAuthConfigSpecification);
+  }
+
+  /**
+   * Returns the URL where to retrieve the access token from.
+   *
+   */
+  @Override
+  protected String getAccessTokenUrl() {
```

Comment:
```suggestion
  @Override
  protected String getAccessTokenUrl() {
    throw new IOException("not supported");
  }
  
  @Override
  protected String getAccessTokenUrl(final JsonNode inputOAuthConfiguration) {
```

This method should be added in `airbyte-oauth/src/main/java/io/airbyte/oauth/BaseOAuth2Flow.java` to also include the config object to the call.

- All Oauth flow implementation that are not depending on a value from connector config should call the empty `getAccessTokenUrl`
- All Oauth flow implementation that are depending on a value from connector config should throw exceptions when calling the empty `getAccessTokenUrl` method